api/include/mfxdefs.h

```diff
@@ -50,6 +50,41 @@ extern "C"
 {
 #endif /* __cplusplus */
 
+/* In preprocessor syntax # symbol has stringize meaning,
+   so to expand some macro to preprocessor pragma we need to use
+   special compiler dependent construction */
+
+#if defined(_MSC_VER)
+    #define MFX_PRAGMA_IMPL(x) __pragma(x)
+#else
+    #define MFX_PRAGMA_IMPL(x) _Pragma(#x)
+#endif
+
+/* The general rule for alignment is following:
+   - structures with pointers have 4/8 bytes alignment on 32/64 bit systems
+   - structures with fields of tupe mfxU64/mfxF64 (unsigned long long / double)
+     have alignment 8 bytes on 64 bit and 32 bit Windows, on Linux alignment is 4 bytes
+   - all the rest structures are 4 bytes aligned */
```

Comment:
It might help if you will add macro name you intend to use in each of the above 3 cases. Also, might help to highlight what you intend to do in case of structure containing both 1 and 2 from the above?

Reply:
you mean something like this?
```
#define MFX_PACK_USUAL_STRUCT() MFX_PACK_BEGIN_X(4)
#define MFX_PACK_STRUCT_W_PTR() MFX_PACK_BEGIN()
#define MFX_PACK_STRUCT_L_TYPE() MFX_PACK_BEGIN_MAX()
```
(you may suggest better names)
of course we can get rid of ```MFX_PACK_BEGIN``` and ```MFX_PACK_BEGIN_MAX``` is such case

Reply:
implemented in new patch